### PR TITLE
Pierre: ham-lint-fix

### PR DESCRIPTION
**Pull request by prenaux**
Add ham-lint-fix & the added infra to integrate with ham (_build.ham).

Todo:
- [x] bash (shellcheck & shellfmt)
- [x] cpp (clang-format lint & fix)

### CODE REVIEW

**Comment by prenaux** [comment]
Current dependencies on/for this PR:
* master
  * **PR #32** <a href="https://app.graphite.dev/github/pr/prenaux/ham/32" target="_blank">Graphite</a>  👈
    * **PR #33** <a href="https://app.graphite.dev/github/pr/prenaux/ham/33" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/32?utm_source=stack-comment).

File: bin/ham-lint-fix-lib.sh [hunk 1]
```diff
@@ -0,0 +1,420 @@
+#!/bin/bash
+export HAM_LINT_ALL_FILENAME="_lint_all_files.txt"
```
Comment (jamesrdd) [inline bin/ham-lint-fix-lib.sh:2]: Took me a bit to realize what this `_lint_all_files.txt` file was, I think it should be documented somewhere what the file is and the expected format of the text file (even just in a comment here in this script). Something like

```bash
# The `_lint_all_files.txt` contains a line-separated list of linting directives for a module.
# A linting directive is of the form `{LANGUAGE} {RELATIVE/PATH/TO/SRC/DIR}`
```

**Review by jamesrdd — APPROVED** [review]
Looks good. Just a small comment about adding some clarity around the `_lint_all_files.txt` file

File: bin/ham-lint-fix-lib.sh (hunk 1 above)
Comment (prenaux) [reply bin/ham-lint-fix-lib.sh:2]: Noted, I'll document it if it survives the next PR. I'm considering just getting rid of this altogether and doing "per file" only coupled with just using _build.ham to list the files which is just more flexible and not yet another kind of file to worry about.

**Comment by prenaux** [comment]
@prenaux started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/32).

**Comment by prenaux** [comment]
@prenaux merged this pull request with [Graphite](https://app.graphite.dev/github/pr/prenaux/ham/32).

**Pull request merged**